moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -548,8 +544,8 @@ KinematicState Servo::getNextJointState(const moveit::core::RobotStatePtr& robot
     }
   }
 
-  // Update internal state of filter with final calculated command.
-  resetSmoothing(target_state);
+  // Apply smoothing to the positions if a smoother was provided.
+  doSmoothing(target_state);
```

Comment:
Calling resetSmoothing() in the loop was a bug. It's not compatible with the Butterworth filter b/c it wipes out the history of smoothed positions. See https://github.com/moveit/moveit2/blob/f30d51b2cef111ca92c89ced8776960d6e3b2470/moveit_core/online_signal_smoothing/src/butterworth_filter.cpp#L147

```
  for (size_t joint_idx = 0; joint_idx < num_positions; ++joint_idx)
  {
    position_filters_.at(joint_idx).reset(positions[joint_idx]);
  }
```

Reply:
@pac48 can you do this reset at the end of the `doSmoothing()` method of your filter?